static/js/publisher/release/actions/releases.js

```diff
@@ -1,5 +1,164 @@
+
+    if (errors.length) {
+      message = `${message} ${errors
+        .map(e => e.message)
+        .filter(m => m)
+        .join(" ")}`;
+    }
+  }
+
+  return showNotification({
+    status: "error",
+    appearance: "negative",
+    content: message
+  });
+}
+
+// TODO: move inside of this function out
+function handleReleaseResponse(
+  dispatch,
+  json,
+  release,
+  revisions,
+  defaultTrack
+) {
+  if (json.success) {
+    // Update channel map based on the response
+    // We need to use channel_map_tree to get branches
+    Object.keys(json.channel_map_tree).forEach(trackKey => {
+      const track = json.channel_map_tree[trackKey];
+      Object.keys(track).forEach(seriesKey => {
+        const series = track[seriesKey];
+        Object.keys(series).forEach(archKey => {
+          const arch = series[archKey];
+          arch.forEach(map => {
+            if (map.revision) {
+              let revision;
+
+              if (map.revision === +release.id) {
+                // release.id is a string so turn it into a number for comparison
+                revision = release.revision;
+              } else if (revisions[map.revision]) {
+                revision = revisions[map.revision];
+              } else {
+                revision = {
+                  revision: map.revision,
+                  version: map.version,
+                  architectures: release.revision.architectures
+                };
+              }
+
+              let channel = map.channel;
+              if (RISKS.indexOf(channel.split("/")[0]) > -1) {
+                channel = `${defaultTrack}/${channel}`;
+              }
+
+              dispatch(releaseRevisionSuccess(revision, channel));
+            }
+          });
+        });
+      });
+    });
+  }
+}
+
+export function releaseRevisions() {
+  return (dispatch, getState) => {
+    const { pendingReleases, pendingCloses, revisions, options } = getState();
+    const { csrfToken, snapName, defaultTrack } = options;
+    const releases = Object.keys(pendingReleases).map(id => {
+      return {
+        id,
+        revision: pendingReleases[id].revision,
+        channels: pendingReleases[id].channels
+      };
+    });
+
+    const _handleReleaseResponse = (json, release) => {
+      return handleReleaseResponse(
+        dispatch,
+        json,
+        release,
+        revisions,
+        defaultTrack
+      );
+    };
+
+    const _handleCloseResponse = json => {
+      return handleCloseResponse(dispatch, json, pendingCloses);
+    };
+
+    dispatch(hideNotification());
+    return fetchReleases(_handleReleaseResponse, releases, csrfToken, snapName)
+      .then(() =>
+        fetchCloses(_handleCloseResponse, csrfToken, snapName, pendingCloses)
+      )
+      .then(() => fetchReleasesHistory(csrfToken, snapName))
+      .then(json => dispatch(updateReleasesData(json)))
+      .catch(error => dispatch(handleReleaseError(error)))
```

Comment:
Now that `handleReleaseError` returns an action the name is a bit confusing (other `handle*` functions are just handlers/callbacks that are not dispatched.

I would at least rename `handleReleaseError` to ... `showErrorNotification` maybe? Or change it so it only returns error message `getErrorMessage(error)` and here do `dispatch(showNotification({ ... message: getErrorMessage(error) })`.